Hmac tests

rprakas-gsc opened the pull request:
Added Hmac error conditions test

rprakas-gsc commented:
@moidx for visibility

nasahlpa commented on sw/device/tests/hmac_error_conditions_test.c:
```diff
@@ -0,0 +1,361 @@
+
+  dif_hmac_t hmac;
+  test_setup(mmio_region_from_addr(TOP_EARLGREY_HMAC_BASE_ADDR), &hmac);
+  for (int error_condition = 1; error_condition < 9; error_condition++) {
+    switch (error_condition) {
```
A for + switch construct seems a bit unnecessary in my opinion.

nasahlpa reviewed (commented):
Thanks for your work @rprakas-gsc! I've dropped a couple of suggestions.

rprakas-gsc commented:
@nasahlpa - Thanks for the suggestions.  The opentitan master test environment is broken.  I will be incorporating these feedback -and checking in this code and filing PR in earlgrey_1.0.0 branch .

nasahlpa commented:
@rprakas-gsc why is the test environment broken on master? I think you just need to restart the CI or rebase your work.

I don't quite understand why you've opened a second, identical PR (#26049)? I'd prefer merging this one as here the review history remains preserved and it is easier for me to do a second review round.

rprakas-gsc commented:
I tried rebasing already on the master branch, it didn't help.  My issue is more related to detecting the FPGA board.  The reason I started an identical PR on the earlgrey_1.0.0 branch, i can test this code on silicon as well.

nasahlpa commented:
Thanks for your response. Not sure if I understand, but your test executes just fine [here in CI](https://github.com/lowRISC/opentitan/actions/runs/12782653333/job/35632684767?pr=25825#step:6:4182):
`//sw/device/tests:hmac_error_conditions_test_fpga_cw340_sival            PASSED in 24.2s`

rprakas-gsc commented:
the tests which was executed in fpga - happened before I rebased.  To fix the merge issues, I rebased it - thats when the FPGA board detection issue started showing up.

rprakas-gsc commented:
Closing this PR - since this PR has been approved in earlgrey_1.0.0 branch and backported to master
